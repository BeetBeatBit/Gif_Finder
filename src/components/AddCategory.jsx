import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [ InputValue, setInputValue ] = useState('')

    /* Para poder cambiar el input, ya que sin el onInputChange se renderiza
      como read only, para poderlo cambiar hay que hacer un onChange como abajo */
    /* Hay que pasarle el evento para que pueda seguir escribiendo también */
    const onInputChange = ( event ) => {
        setInputValue( event.target.value )
    }

    const onSubmit = ( event ) => {
        event.preventDefault()
        if( InputValue.trim().length <= 1) return
        // onAddCategories(categories => [...categories, InputValue])
        onNewCategory( InputValue.trim() )
        setInputValue('')
    }

    return (
        <form onSubmit={ (event) => onSubmit(event)} aria-label='form'>
            <input 
            type = "text"
            placeholder = 'Buscar Gifs'
            value = {InputValue}

            /* Tenemos un caso donde estoy mandando una funcion cuyo primer argumento
            es el argumento que estoy mandándole a la function que quiero ejectutar. Por lo tanto, se puede obviar de la siguiente manera:
            onChange = { onInputChange } */
            onChange = { (event) => onInputChange(event) } 
            />
        </form>  
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
