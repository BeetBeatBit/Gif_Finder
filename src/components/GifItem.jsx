
export const GifItem = ({ title, url }) => { //Recuerda que en los corchetes significa que desestructuramos
  return (
    <div className="card">

        <img src={url} alt={title} />
        <p>{title}</p>
        
    </div>
  )
}