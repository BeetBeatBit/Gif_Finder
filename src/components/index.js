//Este es un archivo de barril
export * from "../components/AddCategory";
export * from "../components/GifGrid";
export * from "../components/GifItem";