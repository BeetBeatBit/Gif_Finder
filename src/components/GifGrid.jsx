import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onRemoveCategory }) => {

  const { images, isLoading } = useFetchGifs(category);
  // console.log({isLoading})
  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<h2>Cargando...</h2>)
      }

      <button onClick = { () => onRemoveCategory(category) }>Eliminar</button>
      
      <div className="card-grid">
        {
          images.map((img) => (

            <GifItem
              key={img.id}
              {...img}
            />

          ))
        }
      </div>
    </>
  )
}
