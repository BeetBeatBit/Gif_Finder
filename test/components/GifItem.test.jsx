import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem Component', () => {
    const title = 'Saitama'
    const url = 'prueba'

    test('Match with Snapshot', () => {
        const { container }= render( <GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })
})