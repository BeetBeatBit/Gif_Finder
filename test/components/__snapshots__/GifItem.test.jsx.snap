// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Component Match with Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="prueba"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
